src/array.vala

```diff
@@ -411,4 +333,93 @@ public class Vast.Array : Object
         }
         return sb.str + (string) @out.steal_data ();
     }
+
+    internal void
+    value_to_memory (Value val, void * memory)
+    {
+        var dest_value = Value (scalar_type);
+        if (val.transform (ref dest_value)) {
+            if (scalar_type == typeof (string)) {
+                var str  = dest_value.get_string ();
+                var dest = Posix.strncpy ((string) memory, str, scalar_size - 1);
+                dest.data[scalar_size - 1] = '\0';
+            }
+
+            else if (dest_value.fits_pointer ()) {
+                Memory.copy (memory, dest_value.peek_pointer(), scalar_size);
+            }
+
+            else if (scalar_type == Type.BOXED) {
+                Memory.copy (memory, dest_value.get_boxed (), scalar_size);
+            }
+
+            else if (scalar_type == typeof (char)) {
+                var _ = dest_value.get_char ();
+                Memory.copy (memory, (&_), scalar_size);
+            }
+
+            else if (scalar_type == typeof (uint8)) {
+                var _ = dest_value.get_uchar ();
+                Memory.copy (memory, (&_), scalar_size);
+            }
+
+            else if (scalar_type == typeof (int64)) {
+                var _ = dest_value.get_int64 ();
+                Memory.copy (memory, (&_), scalar_size);
+            }
+
+            else if (scalar_type == typeof (double)) {
+                var _ = dest_value.get_double ();
+                Memory.copy (memory, &_, scalar_size);
+            }
+
+            else {
+                assert_not_reached ();
+            }
+        } else {
+            error ("Could not transform '%s' into '%s'.", val.type ().name (),
+                                                          dest_value.type ().name ());
+        }
+    }
+
+    internal Value
+    memory_to_value(void * memory)
+    {
+        var _value = Value (scalar_type);
```

Comment:
It's fine to remove the underscore in the variable name as we do not have conflicts anymore.